File: b964(score_index)/Python/score_index.py
from sys import stdin

while True:
    number_of_students = stdin.readline()
    if not number_of_students: break
    number_of_students = int(number_of_students)
    scores = list(map(int, stdin.readline().split()))
    scores.sort()
    print(*scores)
    if scores[number_of_students-1] < 60:
        print(scores[number_of_students-1], 'worst case', sep = '\n')
    elif scores[0] >= 60:
        print('best case', scores[0], sep = '\n')
    else:
        worst, best = 0, 0
        for i in range(number_of_students-1):
            if (scores[i] < 60 and scores[i+1] >= 60):
                worst, best = scores[i], scores[i+1]
        print(worst, best, sep = '\n')

File: b964(score_index)/C++/score_index.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  int number_of_students;
  while (std::cin >> number_of_students) {
    int score;
    std::vector<int> fail, pass;
    for (int s = 0; s < number_of_students; s++) {
      std::cin >> score;
      if (score >= 60) pass.push_back(score);
      else fail.push_back(score);
    }
    std::sort(fail.begin(), fail.end());
    std::sort(pass.begin(), pass.end());

    if (fail.size() == 0) {
      for (int i = 0; i < pass.size()-1; i++) std::cout << pass.at(i) << ' ';
      std::cout << pass.back() << '\n' << "best case\n" << pass.front() << '\n';
    }
    else if (pass.size() == 0) {
      for (int i = 0; i < fail.size()-1; i++) std::cout << fail.at(i) << ' ';
      std::cout << fail.back() << '\n' << fail.back() << '\n' << "worst case\n";
    }
    else {
      for (int s: fail) std::cout << s << ' ';
      for (int i = 0; i < pass.size()-1; i++) std::cout << pass.at(i) << ' ';
      std::cout << pass.back() << '\n' << fail.back() << '\n' << pass.front() << '\n';
    }
  }
  return 0;
}
